index.js

```diff
@@ -177,20 +105,16 @@ module.exports = function (options = {}) {
 			const compiled = compile(code, { ...compilerOptions, filename });
 
 			(compiled.warnings || []).forEach(warning => {
-				if (!css && !emitCss && warning.code === 'css-unused-selector') return;
+				if (!emitCss && warning.code === 'css-unused-selector') return;
 				if (onwarn) onwarn(warning, this.warn);
 				else this.warn(warning);
 			});
 
-			if (external_css && compiled.css.code) {
+			if (emitCss && compiled.css.code) {
 				const fname = id.replace(new RegExp(`\\${extension}$`), '.css');
-
-				if (emitCss) {
-					compiled.js.code += `\nimport ${JSON.stringify(fname)};\n`;
-					compiled.css.code += `\n/*# sourceMappingURL=${compiled.css.map.toUrl()} */`;
-				}
-
-				css_cache.set(fname, compiled.css);
+				compiled.css.code += `\n/*# sourceMappingURL=${compiled.css.map.toUrl()} */`;
```

Comment:
Was talking about this line – we always inline the sourcemap regardless of Rollup config. I think it should be removed entirely, since the `compiled.css` object (kept in cache for `load()`) will pitch the `{ css: string; map?: SourceMap }` contents.

Reply:
Ah, I see. `load` returns `{ css: string; map?: SourceMap }` and the latter plugins (and it's required to have one) will use that `map` attribute. (I originally wasn't quite sure what you meant by "pitch". I was thinking you meant "toss" like "throw away" and was a bit confused, but understand now)

So yeah, I think you're right. We can just delete this line and I think that's the only thing that we'd need to change. It seems somewhat orthogonal to the rest of this change to me still, but I have no objection to including it here since it's just one line. Any reason you haven't updated it yet? I'm happy to make the change and merge this PR to close it out, but wasn't sure if you were wanting to do some more testing of it first or something

Reply:
I was just waiting for consensus on what Rollup plugin(s) we're going to recommend going forward.
Unless we have an answer to that, all Rollup-based Svelte will be left stranded .. or with buggy CSS writers.

Reply:
`rollup-plugin-css-only` seems like a good option now that it uses `emitFile`. `rollup-plugin-postcss` might be a bit heavy-weight to include by default. I think you may have mentioned another option? If so, it'd be good to include it in the README in this PR where you added a mention of `rollup-plugin-css-only` and `rollup-plugin-postcss`